RecoTracker/TkDetLayers/src/Phase2OTtiltedBarrelLayer.cc

```diff
@@ -0,0 +1,150 @@
+  LogDebug("TkDetLayers") << "==== DEBUG Phase2OTtiltedBarrelLayer =====" ; 
+  LogTrace("TkDetLayers") << "Phase2OTtiltedBarrelLayer Cyl r,lenght: "
+                          << theCylinder->radius() << " , "
+                          << theCylinder->bounds().length();
+
+  for (vector<const GeometricSearchDet*>::const_iterator i=theNegativeRingsComps.begin();
+       i != theNegativeRingsComps.end(); i++){
+    LogTrace("TkDetLayers") << "negative rings in Phase2OT tilted barrel pos z,perp,eta,phi: " 
+			    << (**i).position().z()    << " , " 
+			    << (**i).position().perp() << " , " 
+			    << (**i).position().eta()  << " , " 
+			    << (**i).position().phi()  ;
+  }
+  
+  for (vector<const GeometricSearchDet*>::const_iterator i=Phase2OTBarrelLayer::components().begin();
+       i != Phase2OTBarrelLayer::components().end(); i++){
+    LogTrace("TkDetLayers") << "rods in Phase2OT tilted barrel pos z,perp,eta,phi: " 
+			    << (**i).position().z()    << " , " 
+			    << (**i).position().perp() << " , " 
+			    << (**i).position().eta()  << " , " 
+			    << (**i).position().phi()  ;
+  }
+
+  for (vector<const GeometricSearchDet*>::const_iterator i=thePositiveRingsComps.begin();
+       i != thePositiveRingsComps.end(); i++){
+    LogTrace("TkDetLayers") << "positive rings in Phase2OT tilted barrel pos z,perp,eta,phi: " 
+			    << (**i).position().z()    << " , " 
+			    << (**i).position().perp() << " , " 
+			    << (**i).position().eta()  << " , " 
+			    << (**i).position().phi()  ;
+  }
+  LogTrace("TkDetLayers") << "==== end DEBUG Phase2OTtiltedBarrelLayer =====" ; 
+  //----------------------------------- 
+
+}
+
+Phase2OTtiltedBarrelLayer::~Phase2OTtiltedBarrelLayer(){
+
+  vector<const GeometricSearchDet*>::const_iterator i;
+  for (i=theNegativeRingsComps.begin(); i!=theNegativeRingsComps.end(); i++) {
+    delete *i;
+  }
+  for (i=thePositiveRingsComps.begin(); i!=thePositiveRingsComps.end(); i++) {
+    delete *i;
+  }
+
+} 
+
+void 
+Phase2OTtiltedBarrelLayer::groupedCompatibleDetsV( const TrajectoryStateOnSurface& tsos,
+					  const Propagator& prop,
+					   const MeasurementEstimator& est,
+					   std::vector<DetGroup> & result) const {
+  vector<DetGroup> closestResultRods;
+  vector<DetGroup> closestResultNeg;
+  vector<DetGroup> closestResultPos;
+  Phase2OTBarrelLayer::groupedCompatibleDetsV(tsos, prop, est, closestResultRods);
+  for(auto ring : theNegativeRingsComps){
+    ring->groupedCompatibleDetsV(tsos, prop, est, closestResultNeg);
+  }
+  for(auto ring : thePositiveRingsComps){
+    ring->groupedCompatibleDetsV(tsos, prop, est, closestResultPos);
+  }
+
+  result.assign(closestResultRods.begin(),closestResultRods.end());
+  result.insert(result.end(),closestResultPos.begin(),closestResultPos.end());
+  result.insert(result.end(),closestResultNeg.begin(),closestResultNeg.end());
+
+  LogDebug("TkDetLayers") << "==== output di Phase2OTtiltedBarrelLayer =====" ; 
```

Comment:
everything below is debugs not compiled anyways except for a bunch of empty loops
put this inside ```#ifdef EDM_ML_DEBUG ```